sound/soc/sof/sof-priv.h

```diff
@@ -26,6 +26,7 @@
 #define SOF_DBG_TEXT		BIT(3)
 #define SOF_DBG_PCI		BIT(4)
 #define SOF_DBG_RETAIN_CTX	BIT(5)	/* prevent DSP D3 on FW exception */
+#define SOF_DBG_FORCE_ERR_LEVEL	BIT(6) /* used to dump dsp status with error log level */
```

Comment:
`SOF_DBG_*` macros are used for two purposes:
1. for the `sof_core_debug` module parameter, in which case only two values are checked: `SOF_DBG_ENABLE_TRACE` and `SOF_DBG_RETAIN_CTX`
2. in the `flags` parameter to `ops.dbg_dump()` where they aren't ever checked.

Just sayin' :-)

Reply:
@ranj063 Do you remember history with this? It does seem fairly odd, sof-priv.h clearly states "global debug state set by SOF_DBG_ flags" for sof_core_debug. 

In snd_sof_handle_fw_exception(), a single small function uses the same bits for two different purposes.

I think we should just clean this up and have debug dump flags in their own enumeration. This is now very confusing for someone trying to understand how to use "sof_core_debug" module parameter.